interpreter/exec/eval.ml

```diff
@@ -510,10 +565,70 @@ let rec step (c : config) : config =
     | Frame (n, frame', (vs', {it = Returning vs0; at} :: es')), vs ->
       take n vs0 e.at @ vs, []
 
+    | Frame (n, frame', (vs', {it = Throwing (a, vs0); at} :: es')), vs ->
+      vs, [Throwing (a, vs0) @@ at]
+
     | Frame (n, frame', code'), vs ->
       let c' = step {frame = frame'; code = code'; budget = c.budget - 1} in
       vs, [Frame (n, c'.frame, c'.code) @@ e.at]
 
+    | Catch (n, cts, ca, (vs', [])), vs ->
+      vs' @ vs, []
+
+    | Catch (n, cts, ca, (vs', {it = Delegating (0l, a, vs0); at} :: es')), vs ->
+      vs, [Catch (n, cts, ca, (vs', (Throwing (a, vs0) @@ at) :: es')) @@ e.at]
```

Comment:
Since delegation always targets a label, this rule should be unnecessary.

Reply:
This rule seems needed as tests fail without it. I believe the issue is that since the label is outside the catch (when `TryCatch` is reduced to `[Label (... [Catch ...])]`), if we drop the `Catch` and let `Delegating` go up to the label, the handler that we were supposed to delegate to is gone.

Reply:
I see. Hm, this is a bit unfortunate. Operationally, we could just switch around the nesting of Label and Catch to handle this naturally, but that would break the hack we use to type catch labels... Fine for now I guess, maybe I can come up with an idea later.

@ioannad, FYI, since @takikawa's observation likely applies to the formal reduction rules as well.